Support settings roles by default on authentication

Sebobo opened the pull request.

Sebobo committed:
[BUGFIX] Configuration loads when providername contains a dot

Sebobo committed:
[TASK] Support setting roles by default

With this change you don't need to have
your own implementation of the LDAPProvider
anymore.
Only changing the configuration is enough.

radmiraal commented on Configuration/Settings.yaml:
```diff
@@ -1,4 +1,11 @@
\ No newline at end of file
+    allowStandinAuthentication: FALSE
+    roles:
+      default:
```
shouldn't this be an empty array too?

radmiraal commented on Classes/TYPO3/LDAP/Security/Authentication/Provider/LDAPProvider.php:
```diff
@@ -173,6 +215,63 @@ protected function updateParty(Account $account, array $ldapSearchResult) {
+			}
+
+			$dn = $this->eelEvaluator->evaluate($this->partyConfiguration['dn'], $eelContext);
+
```
if you're on fixing that other comment about the empty array in the config file I'd remove this empty line (236) too

Sebobo committed:
[TASK] Catch ldap errors and throw exceptions

Sebobo committed:
[BUGFIX] Bind with admin after user bind

The group search can fail when a user
doesn't have the permission to search for group
memberships.
With this change the admin is bound again after
the user bind.

Also a second user search is not made when
we already have the results.

Sebobo commented:
Hey @radmiraal can you check this PR again? 
I accidentally added two more changes ;)

radmiraal commented:
thnx ;) I need to do an LDAP / Active Directory SSO setup within the next 2 weeks, will give the change a spin while doing so

radmiraal commented:
Found a few minor tweaks which I'll push in another change, but this definitely works / looks good ;-) thanks!
